packages/components/Callout/macos/CalloutWindow.swift

```diff
@@ -22,11 +22,36 @@ class CalloutWindow: NSWindow {
 		// Dismiss the Callout if the window is no longer active.
 		NotificationCenter.default.addObserver(self, selector: #selector(dismissCallout), name: NSApplication.didResignActiveNotification, object: nil)
 
-		// Dismiss the Callout if the user touched/clicked outside the bounds of the callout.
+		// Dismiss the Callout if the user touched/clicked outside the Callout Window or any of our child windows.
 		mouseEventMonitor = NSEvent.addLocalMonitorForEvents(matching: .leftMouseUp, handler: { [weak self] (event) -> NSEvent? in
-			if (event.window != self) {
+			guard let clickedWindow = event.window else {
+				return event
+			}
+
+			var shouldDismissCallout = false
+
+			if (clickedWindow.isEqual(to: self) ) {
+				shouldDismissCallout = false
+			} else if (clickedWindow.isMainWindow) {
+				shouldDismissCallout = true
+			} else if (clickedWindow.isEqual(to: self?.parent) && self?.parent is CalloutWindow) {
+				// We are a child window of a Callout (e.g: a ContextualMenu Submenu), and the click happened in our parent
+				shouldDismissCallout = false
+			} else {
+				// Check if the click happened in any of our child windows (e.g: our submenus)
+				if let childWindows = self?.childWindows {
```

Comment:
childWindows ... with a "s" it sounds like it is an array?

Reply:
It is an array. You can have multiple childwindows.

Reply:
ah thansk. something about line 42 and line 43 doesn't feel very swifty. self?.childWindws.contain(clickedwindow) ?? true  ?

Reply:
That's fair. The checks with optional Bools (`Bool?`) trip me up. I tried using the `if let` to be more swifty, but I think the nil coalescing operator might be cleaner overall!

Reply:
It ended up needing to be `self?.childWindows.contains(clickedwindow) ?? false ` but I like this better! Thanks =)